hpx/runtime/threads/policies/thread_queue.hpp

```diff
@@ -196,8 +196,8 @@ namespace hpx { namespace threads { namespace policies
         typedef typename PendingQueuing::template
             apply<thread_description*>::type work_items_type;
 
-        typedef typename StagedQueuing::template
-            apply<task_description*>::type task_items_type;
+        typedef lockfree_fifo::
```

Comment:
Either `StagedQueueing` should go from the template, or the other queues should still be supported?

Reply:
StagedQueuing is still used for the thread_alloc_ template.
Not sure if it's actually needed though.

Reply:
It's not actually used, so it could go.

Reply:
I'll clean it up some more then.

Do @hkaiser or @sithhell want to comment on whether we should move lockfree stuff into concurrent, or just move the concurrentqueue into util/lockfree for now. (seems like putting it into util/lockfree for the time being is probably the simplest option, though I still like the idea of having a concurrent top level dir for this stuff)

Reply:
I'm impartial to this. Having it in util/lockfree is fine, I guess.